app/uk/gov/hmrc/fileupload/read/envelope/Repository.scala

```diff
@@ -48,56 +48,63 @@ object Repository {
 
   val deleteSuccess = Right(DeleteSuccess)
 
-  def apply(mongo: () => DB with DBMetaCommands): Repository = new Repository(mongo)
+  def apply(mongoComponent: MongoComponent)(implicit ec: ExecutionContext): Repository = new Repository(mongoComponent)
 }
 
-class Repository(mongo: () => DB with DBMetaCommands)
-  extends ReactiveRepository[Envelope, BSONObjectID](collectionName = "envelopes-read-model", mongo, domainFormat = Envelope.envelopeFormat) {
+class Repository(mongoComponent: MongoComponent)
+                (implicit ec: ExecutionContext)
+  extends PlayMongoRepository[Envelope](
+    collectionName = "envelopes-read-model",
+    mongoComponent = mongoComponent,
+    domainFormat = Envelope.envelopeFormat,
+    indexes = Seq(
+      IndexModel(Indexes.ascending("status", "destination"), IndexOptions().background(true))
+    )) {
 
-  val duplicateKeyErrorCode = Some(11000)
-
-  override def ensureIndexes(implicit ec:ExecutionContext) = {
-    collection.indexesManager.ensure(Index(key = List("status" -> IndexType.Ascending, "destination" -> IndexType.Ascending), background = true)).map(Seq(_))
-  }
-
-  import reactivemongo.play.json.ImplicitBSONHandlers._
   import Repository._
 
-  def update(writeConcern: WriteConcern = WriteConcern.Default)
-            (envelope: Envelope, checkVersion: Boolean = true)
-            (implicit ex: ExecutionContext): Future[UpdateResult] = {
+  def update(writeConcern: WriteConcern = WriteConcern.MAJORITY)
```

Comment:
was the `WriteConcern.Default` actually `WriteConcern Majority`?

Reply:
No, and yes... :)

Default was a single Acknowledged, but this code overrode that for most cases, and set N=2:
https://github.com/hmrc/file-upload/blob/master/app/uk/gov/hmrc/fileupload/ApplicationModule.scala#L94

Decided it would be worth removing that logic, and just setting to Majority.
In the case of our environments, the effect would be the same. Running locally, majority is 1. Plus its far simpler.

Also, the new driver doesn't have "Default" as an option.